src/pages/ReimbursementAccount/BeneficialOwnersStep.js

```diff
@@ -80,7 +83,15 @@ class BeneficialOwnersStep extends React.Component {
     }
 
     removeBeneficialOwner(beneficialOwner) {
-        this.setState(prevState => ({beneficialOwners: _.without(prevState.beneficialOwners, beneficialOwner)}));
+        this.setState((prevState) => {
+            const beneficialOwners = _.without(prevState.beneficialOwners, beneficialOwner);
+
+            // We set 'beneficialOwners' to null first because we don't have a way yet to replace a specific property without merging it.
+            // We don't use the debounced function because we want to make both function calls.
+            updateReimbursementAccountDraft({beneficialOwners: null});
+            updateReimbursementAccountDraft({beneficialOwners});
```

Comment:
Can you add some tests to the description to confirm this update works as expected?

Reply:
Updated the description with testing steps for the potential race condition.